src/pg/pg-store.ts

```diff
@@ -229,6 +229,9 @@ export class PgStore extends BasePgStore {
           await this.brc20.scanBlocks(event.block_identifier.index, event.block_identifier.index);
       }
     });
+    if (payload.rollback.length > 0) {
+      await this.brc20.normalizeDeployTxCounts(); // not efficient re-count, but only happens on rollbacks
```

Comment:
I think this won't fly 😢 the activities table is already a dozen million records so this will take a very long time. We should traverse the rollback txs and discount their values from the column total. This could be tricky because you'll need to check if the tx did generate an activity before subtracting 1

Reply:
Yeah, I thought so. I was thinking we could do some other optimizations like only re-count touched deploys (which would still be a lot, but a lot less than all).
tx_count should be doable either way. But it will especially get annoying if we add something similar for holder_count and then need to take care of detecting zero balances. etc. getting far away from just deleting locations and cascading 😞

Reply:
One approach might be to add more block by block data, and recompute starting from a "snapshot" similar to the stats endpoint. But that will add some data overhead 🤷‍♂

Reply:
I'm working on some ideas for token count management right now and looking at rollback handling... I'll send you my progress once I have something so we can brainstorm from there